tests/e2e/dualstack/dualstack_test.go

```diff
@@ -197,7 +197,28 @@ var _ = Describe("Verify DualStack Configuration", Ordered, func() {
 			}, "10s", "1s").Should(ContainSubstring("ds-nodeport-pod"), "failed cmd: "+cmd)
 		}
 	})
-
+	It("Verifies podSelector Network Policy", func() {
+		_, err := e2e.DeployWorkload("pod_client.yaml", kubeConfigFile)
+		Expect(err).NotTo(HaveOccurred())
+		cmd := "kubectl exec svc/client-curl --kubeconfig=" + kubeConfigFile + " -- curl -m7 ds-clusterip-svc/name.html"
+		Eventually(func() (string, error) {
+			return e2e.RunCommand(cmd)
+		}, "20s", "3s").Should(ContainSubstring("ds-clusterip-pod"), "failed cmd: "+cmd)
+		_, err = e2e.DeployWorkload("netpol-fail.yaml", kubeConfigFile)
+		Expect(err).NotTo(HaveOccurred())
+		cmd = "kubectl exec svc/client-curl --kubeconfig=" + kubeConfigFile + " -- curl -m7 ds-clusterip-svc/name.html"
+		Eventually(func() error {
+			_, err = e2e.RunCommand(cmd)
+			Expect(err).To(HaveOccurred())
+			return err
+		}, "20s", "3s")
```

Comment:
```suggestion
		Eventually(func(g Gomega) {
			_, err := e2e.RunCommand(cmd)
			g.Expect(err).To(HaveOccurred())
			return err
		}, "20s", "3s")
```
You need to use the `g Gomega` to be a local check contained within the Eventual loop.

Reply:
Also you might not need the `Eventually` at all, this will likely fail immediately, since you want an error.